Userland/Libraries/LibWeb/HTML/History.cpp

```diff
@@ -44,14 +44,71 @@ WebIDL::ExceptionOr<void> History::replace_state(JS::Value data, String const&,
     return shared_history_push_replace_state(data, url, IsPush::No);
 }
 
+// https://html.spec.whatwg.org/multipage/history.html#dom-history-length
+WebIDL::ExceptionOr<u64> History::length() const
+{
+    // 1. If this's associated Document is not fully active, then throw a "SecurityError" DOMException.
+    if (!m_associated_document->is_fully_active())
+        return WebIDL::SecurityError::create(realm(), "Cannot perform length on a document that isn't fully active."sv);
+
+    // 2. Return the number of entries in the top-level browsing context's joint session history.
+    auto const* browsing_context = m_associated_document->browsing_context();
+    return browsing_context->session_history().size();
```

Comment:
If we're don't have the concept of "joint session history" https://html.spec.whatwg.org/multipage/history.html#joint-session-history implemented in LibWeb yet, then this should have a FIXME to use the joint session history.

> The joint session history of a [top-level browsing context](https://html.spec.whatwg.org/multipage/browsers.html#top-level-browsing-context) is the union of all the [session histories](https://html.spec.whatwg.org/multipage/history.html#session-history) of all [browsing contexts](https://html.spec.whatwg.org/multipage/browsers.html#concept-document-bc) of all the [fully active](https://html.spec.whatwg.org/multipage/browsers.html#fully-active) [Document](https://html.spec.whatwg.org/multipage/dom.html#document) objects that share that [top-level browsing context](https://html.spec.whatwg.org/multipage/browsers.html#top-level-browsing-context), with all the entries that are [current entries](https://html.spec.whatwg.org/multipage/history.html#current-entry) in their respective [session histories](https://html.spec.whatwg.org/multipage/history.html#session-history) removed except for the [current entry of the joint session history](https://html.spec.whatwg.org/multipage/history.html#current-entry-of-the-joint-session-history).

Reply:
ok